articles/spring-data-mongodb/index.md

```diff
@@ -0,0 +1,170 @@
+        return ResponseEntity(todoService.updateTodo(todo), HttpStatus.OK)
+    }
+
+    //delete todo
+    @DeleteMapping("/{id}")
+    fun deleteTodo(@PathVariable("id") id: String): ResponseEntity<Any> {
+        return ResponseEntity(todoService.deleteTodo(id), HttpStatus.NO_CONTENT)
+    }
+
+}
+```
+- `@CrossOrigin` annotation is used to configure the allowed origins.
+- `@RestController` annotation is used to define a controller and to indicate that the return value of the methods should be bound to the web response body.
+- `@RequestMapping` declares that all APIs URL in the controller will start with `/api/v1/todos/`.
+
+### Repository Layer
+In the root package of our application create a package named `repository` and create a kotlin file named `TodoRepository.kt`  within the `repository` package, add the code snippet below.
+```kotlin
+interface TodoRepository : MongoRepository<Todo, String> {
+}
+```
+- `interface TodoRepository` interface extends the `MongoRepository` interface that contains all the CRUD functions that we will use to query our MongoDB collection.
+  
+### Service layer
+In the root project package, create a package name `service` and add two kotlin files `TodoService.kt`, `TodoServiceImpl` with the code snippets below respectively.
+
+```kotlin
+interface TodoService {
+    fun saveTodo(todo: Todo)
+
+    fun getTodos(): List<Todo>
+
+    fun getTodoById(id: String): Todo
+
+    fun deleteTodo(id: String)
+
+    fun updateTodo(todo: Todo)
+}
+```
+
+
+```kotlin
+@Service
+class TodoServiceImpl(val todoRepository: TodoRepository) : TodoService {
+    override fun saveTodo(todo: Todo) {
+        todoRepository.save(todo)
+    }
+
+    override fun getTodos(): List<Todo> {
+        return todoRepository.findAll()
+    }
+
+    override fun getTodoById(id: String): Todo {
+        val todoQueried: Optional<Todo> = todoRepository.findById(id)
+        if (todoQueried.isPresent) {
+            return todoQueried.get()
+        } else {
+            throw IllegalStateException("Todo with id $id not found")
+        }
+    }
+
+    override fun deleteTodo(id: String) {
+        todoRepository.deleteById(id)
+    }
+
+    override fun updateTodo(todo: Todo) {
+        val todoQueried: Optional<Todo> = todoRepository.findById(todo.id)
+        if (todoQueried.isPresent) {
+            val _todo = todoQueried.get()
+            _todo.description = todo.description
+            _todo.isCompleted = todo.isCompleted
+            _todo.title = todo.title
+            todoRepository.save(_todo)
+        } else {
+            throw IllegalStateException("Todo with id ${todo.id} is not found")
+        }
+    }
+}
+```
+### Configurations
+In the resources directory, open the `application.properties` file in your IDE and add the configurations below.
+```yaml
+spring.data.mongodb.database=todo_db
+spring.data.mongodb.port=27017
+
+```
+- `spring.data.mongodb.database=todo_db` indicates that our database name is `todo_db`.
```

Comment:
It would be great if you wrote on how to connect the application to the online mongodb clusters